src/contexts/states/logFileStore.ts

```diff
@@ -0,0 +1,160 @@
+                    " structured logs by customizing how fields are displayed.",
+        timeoutMillis: LONG_AUTO_DISMISS_TIMEOUT_MILLIS,
+        title: "A format string has not been configured",
+    });
+};
+
+/**
+ * Handle export progress and append chunks to log file.
+ *
+ * @param logs
+ */
+const handleExportChunk = (logs: string) => {
+    const {logExportManager} = useLogExportStore.getState();
+    if (null !== logExportManager) {
+        const progress = logExportManager.appendChunk(logs);
+        useLogExportStore.getState().setExportProgress(progress);
+    }
+};
+
+/**
+ * Handle query results by updating the query progress and merging the results.
+ *
+ * @param progress
+ * @param results
+ */
+const handleQueryResults = (progress: number, results: QueryResults) => {
+    const {clearQueryResults, setQueryProgress, mergeQueryResults} = useQueryStore.getState();
+
+    if (0 === progress) {
+        clearQueryResults();
+
+        return;
+    }
+    setQueryProgress(progress);
+    mergeQueryResults(results);
+};
+
+const useLogFileStore = create<LogFileState>((set, get) => ({
+    ...LOG_FILE_STORE_DEFAULT,
+    loadFile: (fileSrc: FileSrcType, cursor: CursorType) => {
+        const {setUiState} = useUiStore.getState();
+        const {isPrettified} = useViewStore.getState();
+        const {setFileName, setOnDiskFileSizeInBytes} = get();
+        setUiState(UI_STATE.FILE_LOADING);
+        setFileName(LOG_FILE_STORE_DEFAULT.fileName);
+        useViewStore.getState().setLogData(VIEW_STORE_DEFAULT.logData);
+        setOnDiskFileSizeInBytes(LOG_FILE_STORE_DEFAULT.onDiskFileSizeInBytes);
+
+        useQueryStore.getState().clearQuery();
+        useLogExportStore.getState().setExportProgress(LOG_EXPORT_STORE_DEFAULT.exportProgress);
+
+        set({fileSrc});
+        if ("string" !== typeof fileSrc) {
+            updateWindowUrlSearchParams({[SEARCH_PARAM_NAMES.FILE_PATH]: null});
+        }
+
+
+        const decoderOptions = getConfig(CONFIG_KEY.DECODER_OPTIONS);
+        (async () => {
+            const {fileInfo, pageData} = await useLogFileManagerStore
+                .getState()
+                .logFileManagerProxy
+                .loadFile(
+                    {
+                        cursor: cursor,
+                        decoderOptions: decoderOptions,
+                        fileSrc: fileSrc,
+                        isPrettified: isPrettified,
+                        pageSize: getConfig(CONFIG_KEY.PAGE_SIZE),
+                    },
+                    Comlink.proxy(handleExportChunk),
+                    Comlink.proxy(handleQueryResults)
+                );
+
+            set(fileInfo);
+            useViewStore.getState().updatePageData(pageData);
+
+            if (fileInfo.isStructuredLog && 0 === decoderOptions.formatString.length) {
+                postFormatPopup();
+            }
+        })().catch((reason: unknown) => {
+            useContextStore.getState().postPopUp({
```

Comment:
Can we `console.error(e)` the exception in case `String(e)` doesn't show enough information for debugging?